### PR TITLE
Switch to jquery-datatables

### PR DESCRIPTION
**Pull request by coolo**
jquery-datables-rails is outdated and no longer maintained,
jquery-datables offers an updated version containing bootstrap4
integration

Co-authored-by: Stasiek Michalski <hellcp@opensuse.org>

### CODE REVIEW

**Comment by coolo** [comment]
Can we teach codeclimate somehow to ignore webui2? Otherwise basicall all webui2 PRs will be red :(

**Comment by hennevogel** [comment]
only the ones that introduce duplicated assets. Maybe we should get a *shared* folder going for things we want to keep?

**Comment by coolo** [comment]
the added datatables.js is not duplicated. But its content is very similiar

**Comment by coolo** [comment]
https://github.com/openSUSE/open-build-service/pull/5460 - for the config to ignore webui2 for the moment

**Comment by coolo** [comment]
codeclimate cooled down

File: src/api/app/assets/javascripts/webui/application/datatables.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// require datatables/extensions/RowReorder/dataTables.rowReorder
+// require datatables/extensions/Scroller/dataTables.scroller
+// require datatables/extensions/Select/dataTables.select
+
+//Global setting and initializer
+
+$.extend( $.fn.dataTable.defaults, {
+  responsive: true,
+  pagingType: 'full',
+  //dom:
```
Comment (hennevogel) [inline src/api/app/assets/javascripts/webui/application/datatables.js]: Please remove this boilerplate. If I want to know what this does I have to look it up anyway.

File: src/api/app/assets/javascripts/webui/application/datatables.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+// require datatables/extensions/Buttons/buttons.flash
+// require datatables/extensions/ColReorder/dataTables.colReorder
+// require datatables/extensions/FixedColumns/dataTables.fixedColumns
+// require datatables/extensions/FixedHeader/dataTables.fixedHeader
+// require datatables/extensions/KeyTable/dataTables.keyTable
+// require datatables/extensions/Responsive/dataTables.responsive
+// require datatables/extensions/RowGroup/dataTables.rowGroup
+// require datatables/extensions/RowReorder/dataTables.rowReorder
+// require datatables/extensions/Scroller/dataTables.scroller
+// require datatables/extensions/Select/dataTables.select
```
Comment (hennevogel) [inline src/api/app/assets/javascripts/webui/application/datatables.js]: No need to have this in the code. It'll outdate rahter quickly...

File: src/api/app/assets/javascripts/webui2/datatables.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+// require datatables/extensions/Responsive/dataTables.responsive
+// require datatables/extensions/RowGroup/dataTables.rowGroup
+// require datatables/extensions/RowReorder/dataTables.rowReorder
+// require datatables/extensions/Scroller/dataTables.scroller
+// require datatables/extensions/Select/dataTables.select
+
+//= require datatables/dataTables.bootstrap4
+// require datatables/extensions/AutoFill/autoFill.bootstrap4
+// require datatables/extensions/Buttons/buttons.bootstrap4
+// require datatables/extensions/Responsive/responsive.bootstrap4
```
Comment (hennevogel) [inline src/api/app/assets/javascripts/webui2/datatables.js]: Same as above.

File: src/api/app/assets/javascripts/webui2/datatables.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+//Global setting and initializer
+
+$.extend( $.fn.dataTable.defaults, {
+  responsive: true,
+  pagingType: 'full',
+  //dom:
```
Comment (hennevogel) [inline src/api/app/assets/javascripts/webui2/datatables.js]: Same as above.

File: src/api/app/assets/javascripts/webui2/datatables.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  api = new $.fn.dataTable.Api(settings);
+  tableId = "#" + api.table().node().id;
+  url = $(tableId).data('source');
+  if (url) {
+    return api.ajax.url(url);
+  }
+});
+
+
+// init on turbolinks load
```
Comment (hennevogel) [inline src/api/app/assets/javascripts/webui2/datatables.js]: Same as above.

File: src/api/app/assets/stylesheets/webui/application/datatables.scss [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+//@import 'datatables/extensions/Buttons/buttons.dataTables';
+//@import 'datatables/extensions/ColReorder/colReorder.dataTables';
+//@import 'datatables/extensions/FixedColumns/fixedColumns.dataTables';
+//@import 'datatables/extensions/FixedHeader/fixedHeader.dataTables';
+//@import 'datatables/extensions/KeyTable/keyTable.dataTables';
+//@import 'datatables/extensions/Responsive/responsive.dataTables';
+//@import 'datatables/extensions/RowGroup/rowGroup.dataTables';
+//@import 'datatables/extensions/RowReorder/rowReorder.dataTables';
+//@import 'datatables/extensions/Scroller/scroller.dataTables';
+//@import 'datatables/extensions/Select/select.dataTables';
```
Comment (hennevogel) [inline src/api/app/assets/stylesheets/webui/application/datatables.scss]: No need to have this in the code. It'll outdate rather quickly.

File: src/api/app/assets/stylesheets/webui2/datatables.scss [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+//@import 'datatables/extensions/Buttons/buttons.bootstrap4';
+//@import 'datatables/extensions/ColReorder/colReorder.bootstrap4';
+//@import 'datatables/extensions/FixedColumns/fixedColumns.bootstrap4';
+//@import 'datatables/extensions/FixedHeader/fixedHeader.bootstrap4';
+//@import 'datatables/extensions/KeyTable/keyTable.bootstrap4';
+//@import 'datatables/extensions/Responsive/responsive.bootstrap4';
+//@import 'datatables/extensions/RowGroup/rowGroup.bootstrap4';
+//@import 'datatables/extensions/RowReorder/rowReorder.bootstrap4';
+//@import 'datatables/extensions/Scroller/scroller.bootstrap4';
+//@import 'datatables/extensions/Select/select.bootstrap4';
```
Comment (hennevogel) [inline src/api/app/assets/stylesheets/webui2/datatables.scss]: Same here...

File: src/api/app/assets/javascripts/webui/application/datatables.js (hunk 1 above)
Comment (hennevogel) [reply src/api/app/assets/javascripts/webui/application/datatables.js]: source

**Comment by coolo** [comment]
Removed their inline documentation

**Pull request merged**